content/advanced/430_emr_on_eks/eks_spark_new_york_taxidataset.md

```diff
@@ -0,0 +1,225 @@
+---
+title: "Example - Running Spark jobs on NY Taxi Data"
+date: 2021-05-19T10:30:14-04:00
+weight: 61
+draft: true
+---
+
+In this chater, you will use spark jobs running on EKS to analyze the New York City Taxi and Limousine Commision (TLC) Trip Record Data . The data is available at Registry of Open Data on AWS at https://registry.opendata.aws/nyc-tlc-trip-records-pds/ . For calendar years 2019 and 2020, you will calculate top-10 pickup locations in New York area based on total amount of distance travelled by the taxi's.  
+
+
+You can find more details about Yellow Cabs and Green Cabs at https://www1.nyc.gov/site/tlc/businesses/yellow-cab.page and https://www1.nyc.gov/site/tlc/businesses/green-cab.page.
+
+
+### Creating Nodegroup
+
+Amazon EKS managed node groups automate the provisioning and lifecycle management of nodes (Amazon EC2 instances) for Amazon EKS Kubernetes clusters. All managed nodes are provisioned as part of an Amazon EC2 Auto Scaling group that's managed for you by Amazon EKS. All resources including the instances and Auto Scaling groups run within your AWS account. Each node group run across multiple Availability Zones that you define. 
+
+We always recommend our customers to use multiple AZs (Availability Zones) for building services with high availability and fault tolerance. Sometimes for certain analytics scenarios, where you are processing vast amounts of data in Spark and wish to minimize the inter-AZ data transfer charges between Spark executors, you may provision your managed nodegroup in a single AZ.
+
+Lets create a managed nodegroup called emr-ny-taxi. Please note that for this example all the nodes will be provisioned in a single zone in a private subnet. 
+
+
+```sh
+eksctl create nodegroup --cluster=eksworkshop-eksctl --region=${AWS_REGION} --managed --name=emr-ny-taxi --instance-types=m5.xlarge --nodes-min=3 --nodes-max=3 --node-zones=${AZS[0]} --node-private-networking 
```

Comment:
Can you create new nodegroup by appending to addnodegroup.yaml? I think this is better for Infra-as-code as best practice. Also, I'd add cleanup step by deleting nodegroup using this manifest